src/poke_env/environment/abstract_battle.py

```diff
@@ -511,7 +524,25 @@ def parse_message(self, split_message: List[str]):
             pokemon, _ = split_message[2:4]
             self.get_pokemon(pokemon).cant_move()
         elif split_message[1] == "turn":
+
+            # Saving the beginning-of-turn battle state and events as we go into the turn
+            self.observations[self.turn] = self._current_observation
+
             self.end_turn(int(split_message[2]))
+
+            # Create New Observation and record battle state going into the next turn
+            self._current_observation = Observation()
```

Comment:
I can't do this because of circular imports (importing AbstractBattle in Observation, and then importing Observation from AbstractBattle). I can do this a bit more elegantly and define it via an `__init__(self)`

So, something like
`Observation(
    side_conditions=battle.side_conditions,
    ....
  `
  
  What do you think?

Reply:
i think this could work yeah